pyproject.toml

```diff
@@ -46,9 +46,10 @@ dependencies = [
     "psutil >= 7.0",
     "pynacl == 1.2.1",
     "pypeln == 0.4.9",
-    "pyyaml >= 5.0, < 5.4",
-    "requests >= 2.20, < 2.29 ; python_version < '3.13'",  # upper bound is for compatibility with "docker < 8.0"
-    "requests >= 2.20 ; python_version >= '3.13'",
+    "pyyaml >= 5.0, < 5.4 ; python_version <= '3.10'",  # upper bound is for compatibility with "docker-compose"
+    "pyyaml >= 5.4 ; python_version >= '3.11'",
```

Comment:
Pyyaml < 5.4 has security concerns (see security tab).

Do not update for all Python versions at once to minimize blast radius. Updating pyyaml requires to drop deprecated Docker Compose V1.